src/java/org/apache/cassandra/db/filter/RowFilter.java

```diff
@@ -326,10 +349,7 @@ public List<Expression> expressions()
 
         public Iterator<Expression> iterator()
         {
-            List<Expression> allExpressions = new ArrayList<>(expressions);
-            for (FilterElement child : children)
-                allExpressions.addAll(child.expressions);
-            return allExpressions.iterator();
+            return expressions.iterator();
```

Comment:
that is funny. If you `getExpressions`, you'll get expressions from `root` only.
If you iterate, you'll get expressions from the root and its immediate children, but not from all the nodes.

This doesn't make sense.
According to @mike-tr-adamson test coverage we have is sufficient to warrant making this change without fear of regressions.

Reply:
This was one of the reasons Benjamin doesn't like this code. This was done in such a way that the OR support could be added without impacting the existing filtering. It really needs sorting out properly so we exist only with a tree.

Reply:
As such, it was tested in such a way that we know that it works.